### ISSUE TITLE
SSR Streaming / improve Time to First Byte

### ISSUE THREAD

**Issue by Timkor** [nuxt/nuxt.js#4753]
### What problem does this feature solve?

Although there are some caveats using streaming in SSR. 
See https://ssr.vuejs.org/guide/streaming.html#streaming-caveats

It's still possible to stream some important tags in the `<head>` section, such as:
 - `<link rel="dns-prefetch">` tags
 - `<link rel="preconnect">` tags
 - `<link rel="prefetch">` tags
 - `<link rel="preload">` tags

Maybe even more, such as layout styles.

This will improve **page loading time** since the browser has more time to preconnect, preload and prefetch connections and resources.

### What does the proposed changes look like?

 - Change the order of the HTML template so that tags which are not dependent of any dynamic components are in front.
 - Change the [vue-renderer](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-renderer/src/renderer.js) package in a way that it sends part of the HTML as soon as possible.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8460">#c8460</a>)</em></sub></div>

**Comment by MarvinMiles** [comment]
Seems like "preconnect" works that way:

nuxt.config.js:
```
link: [
      {
        rel: "preconnect",
        href: "https://cdn.domain.com/",
        crossorigin: true
      }
]
```

Other tags I have not tested

**Comment by Timkor** [comment]
> Seems like "preconnect" works that way:
> 
> nuxt.config.js:
> 
> ```
> link: [
>       {
>         rel: "preconnect",
>         href: "https://cdn.domain.com/",
>         crossorigin: true
>       }
> ]
> ```
> Other tags I have not tested

Yes, that's true. I actually mean that the Vue Server Renderer send that information before rendering the page.

**Comment by manniL** [comment]
There is also https://github.com/maoberlehner/vue-lazy-hydration / https://github.com/znck/lazy-hydration ☺️

**Comment by hecktarzuli** [comment]
I've never tried it, but you should also be able to do much (if not all) of these as http headers.  That may make them a tick faster.

example: https://developer.mozilla.org/en-US/docs/Web/HTTP/Link_prefetching_FAQ

**Comment by arpadgabor** [comment]
Now that also AWS lambda has [support for streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/), are there any plans for supporting streaming out of the box?

**Comment by Hebilicious** [comment]
> Now that also AWS lambda has [support for streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/), are there any plans for supporting streaming out of the box?

This has been working with Cloudflare workers for a while, and afaik the only framework that supports it is vite-ssr-plugin. Would love to see it in Nuxt. This shouldn't be too hard to implement either under an experimental flag, but maybe this belongs in the Nitro repo and not the Nuxt one ?

**Comment by hecktarzuli** [comment]
I don't see anything official, but I do see some signs.
https://twitter.com/danielcroe/status/1645322235459035137

**Comment by hecktarzuli** [comment]
> Now that also AWS lambda has [support for streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/), are there any plans for supporting streaming out of the box?

I don't see anything official, but I do see some signs.
https://twitter.com/danielcroe/status/1645322235459035137

**Comment by manniL** [comment]
FYI: Out-of-order streaming is something on the list to explore - no ETA though.

**Comment by jd1378** [comment]
I had found this awesome post by Qwik months ago (is short but packed with info):
https://hackmd.io/@qwik/B18-VWayo

just putting it here in, maybe someone find it useful and start working on a PR for it 😄

**Comment by niutech** [comment]
As for out-of-order HTML streaming without JS, read the [lamplightdev blog post](https://lamplightdev.com/blog/2024/01/10/streaming-html-out-of-order-without-javascript/).

**Comment by jdgamble555** [comment]
Is this still in the works?

**Comment by noamr** [comment]
We (chromium) are considering to add support for out-of-order streaming to the web platform, see [explainer](https://github.com/WICG/declarative-partial-updates?tab=readme-ov-file#part-1-patching-out-of-order-streaming).

I wonder if this would be helpful to Nuxt's cause here?
